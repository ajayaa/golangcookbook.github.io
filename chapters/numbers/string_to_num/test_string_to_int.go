package main

import (
	"fmt"
	"os"
	"strconv"
)

func handleError(err error) {
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
}

func main() {
	str := "123"
	/*
		The first argument to ParseInt is the string to be converted.
		The second argument is the base to which the number is to be converted.
		The third argument is the number of bits to which the number should fit.
	*/
	i, err := strconv.ParseInt(str, 10, 64)
	handleError(err)
	fmt.Printf("%T %d\n", i, i)
	str = "invalidString123"
	_, err = strconv.ParseInt(str, 10, 64)
	handleError(err)
}
