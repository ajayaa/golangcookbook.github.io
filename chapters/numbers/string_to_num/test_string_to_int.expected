int64 123
strconv.ParseInt: parsing "invalidString123": invalid syntax
exit status 1
